eos/form-factors/k-lcdas.cc

```diff
@@ -475,6 +581,66 @@ namespace eos
         return psi4T4 + psi4WW;
     }
 
+    double
+    KaonLCDAs::psi4_i(const double & u, const double & mu) const
+    {
+        // strange quark mass
+        const double m_s  = _imp->model->m_s_msbar(mu);
+        const double m_ud = _imp->model->m_ud_msbar(mu) / 2.0;
+
+        const double m_K  = _imp->m_K;
+        const double f_K  = _imp->f_K;
+
+        // Twist 2 Gegenbauer coefficients
+        const double a1K = _imp->a1K(mu);
+        const double a2K = _imp->a2K(mu);
+
+        // Twist 3 coefficients
+        const double rhopK    = power_of<2>((m_s + m_ud) / _imp->m_K); // EOM constraints, cf. [BBL:2006A], cf. eq. (3.12)
+        const double rhomK    = power_of<2>((m_s - m_ud) / _imp->m_K); // identical in the limit m_q -> 0
+        const double eta3K    = _imp->eta3K(mu);
+        const double omega3K  = _imp->omega3K(mu);
+        const double lambda3K = _imp->lambda3K(mu);
+        const double f3K      = _imp->f3K(mu);
+
+        // Twist 4 coefficients
+        const double delta4K  = _imp->delta4K(mu);
+        const double kappa4K  = _imp->kappa4K(mu);
+        const double omega4K  = _imp->omega4K(mu);
+        const double theta1K  = 7.0 / 10.0 * a1K * delta4K;
+        const double theta2K  = -7.0 / 5.0 * a1K * delta4K;
+        const double phi2K    = -7.0 / 20.0 * a1K * delta4K;
+
+        const double u2 = u * u, u3 = u2 * u, u4 = u3 * u, lnu = std::log(u);
+        const double ubar = 1.0 - u, ubar2 = ubar * ubar, ubar3 = ubar2 * ubar, lnubar = std::log(ubar);
+
+        // Gegenbauer polynomials C_n^(1/2)
```

Comment:
Would it be worth defining these polynomials in a separate file in `eos/maths`? E.g. to improve their evaluation which is not optimal here...

Reply:
Yes. Being tracked in #615. Keeping this PR more manageable.